From eecc474aa7020edc73f2ce6f4eb31691ea419bbb Mon Sep 17 00:00:00 2001
From: Caspar Schutijser <caspar@schutijser.com>
Date: Mon, 1 May 2017 18:11:14 +0200
Subject: [PATCH] Enable High-DPI pixmap attribute

Requested by Ken Moore on IRC because "That will fix all the random
icon pixellation on 4K monitors".

Change-Id: I70c3b4346e6519261142ce5c2bbbcb7ac30bd73e
---
 src/Gui/main.cpp                      | 2 ++
 src/Plugins/AbookAddressbook/main.cpp | 1 +
 2 files changed, 3 insertions(+)

diff --git a/src/Gui/main.cpp b/src/Gui/main.cpp
index f7a42c4ad..76736493a 100644
--- a/src/Gui/main.cpp
+++ b/src/Gui/main.cpp
@@ -72,6 +72,8 @@ int main(int argc, char **argv)
     AppVersion::setGitVersion();
     AppVersion::setCoreApplicationData();
 
+    app.setAttribute(Qt::AA_UseHighDpiPixmaps);
+
     app.setWindowIcon(UiUtils::loadIcon(QStringLiteral("trojita")));
 
     QTextStream qOut(stdout, QIODevice::WriteOnly);
diff --git a/src/Plugins/AbookAddressbook/main.cpp b/src/Plugins/AbookAddressbook/main.cpp
index 58541ce25..21bcc8248 100644
--- a/src/Plugins/AbookAddressbook/main.cpp
+++ b/src/Plugins/AbookAddressbook/main.cpp
@@ -112,6 +112,7 @@ int main(int argc, char **argv) {
         return 0;
     }
     QApplication a(argc, argv);
+    a.setAttribute(Qt::AA_UseHighDpiPixmaps);
     QSettings s(QStringLiteral("flaska.net"), QStringLiteral("be.contacts"));
     s.setValue(QStringLiteral("plugin"), QLatin1String("abookaddressbook"));
     Plugins::PluginManager m(nullptr, &s, QStringLiteral("plugin"), QStringLiteral("plugin"));
-- 
GitLab

